java/src/Ice/src/main/java/com/zeroc/Ice/ObjectAdapter.java

```diff
@@ -1449,57 +1431,81 @@ private List<EndpointI> parseEndpoints(String endpts, boolean oaEndpoints) {
             // Get the router's server proxy endpoints and use them as the published endpoints.
-            //
-            endpoints = new ArrayList<>();
-            for (EndpointI endpt : _routerInfo.getServerEndpoints()) {
-                if (!endpoints.contains(endpt)) {
-                    endpoints.add(endpt);
-                }
-            }
+            endpointsArray = _routerInfo.getServerEndpoints();
         } else {
-            //
-            // Parse published endpoints. If set, these are used in proxies
-            // instead of the connection factory Endpoints.
-            //
-            String endpts =
-                    _instance
-                            .initializationData()
-                            .properties
-                            .getProperty(_name + ".PublishedEndpoints");
-            endpoints = parseEndpoints(endpts, false);
-            if (endpoints.isEmpty()) {
-                //
+            // Parse published endpoints. If set, these are used instead of the connection factory
+            // endpoints.
+            var endpointsList =
+                    parseEndpoints(
+                            _instance
+                                    .initializationData()
+                                    .properties
+                                    .getProperty(_name + ".PublishedEndpoints"),
+                            false);
+
+            if (endpointsList.isEmpty()) {
                 // If the PublishedEndpoints property isn't set, we compute the published endpoints
-                // from the OA endpoints, expanding any endpoints that may be listening on
-                // INADDR_ANY
-                // to include actual addresses in the published endpoints.
-                //
-                for (IncomingConnectionFactory factory : _incomingConnectionFactories) {
-                    for (EndpointI endpt : factory.endpoint().expandIfWildcard()) {
-                        //
-                        // Check for duplicate endpoints, this might occur if an endpoint with a DNS
-                        // name
-                        // expands to multiple addresses. In this case, multiple incoming connection
-                        // factories can point to the same published endpoint.
-                        //
-                        if (!endpoints.contains(endpt)) {
-                            endpoints.add(endpt);
+                // from the factory endpoints.
+                endpointsList =
+                        _incomingConnectionFactories.stream()
+                                .map(IncomingConnectionFactory::endpoint)
+                                .collect(Collectors.toList());
+
+                // Remove all loopback endpoints.
+                var endpointsNoLoopback =
+                        endpointsList.stream()
+                                .filter(e -> !e.isLoopback())
+                                .collect(Collectors.toList());
```

Comment:
Unfortunately, there's a chance that the formatter will want to put this all on one line now.
```suggestion
                                .toList();
```

Reply:
Fixed.